multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ExtractBatchedServicesStep.java

```diff
@@ -24,11 +32,14 @@ protected StepPhase executeStep(ProcessContext context) {
         getStepLogger().debug(Messages.EXTRACT_SERVICES_FROM_BATCH);
 
         List<Resource> batchToProcess = context.getVariable(Variables.BATCH_TO_PROCESS);
+        Set<DynamicResolvableParameter> dynamicResolvableParameters = context.getVariable(Variables.DYNAMIC_RESOLVABLE_PARAMETERS);
 
         ServicesCloudModelBuilder servicesCloudModelBuilder = getServicesCloudModelBuilder(context);
         List<CloudServiceInstanceExtended> servicesCalculatedForDeployment = servicesCloudModelBuilder.build(batchToProcess);
         List<CloudServiceInstanceExtended> servicesToCreate = servicesCalculatedForDeployment.stream()
                                                                                              .filter(CloudServiceInstanceExtended::isManaged)
+                                                                                             .map(serviceToCreate -> resolveParametersForLateResolve(serviceToCreate,
```

Comment:
So we resolve the dynamic parameters in Extract...Step? Wouldn't it make more sense to be a separate step with a better name explaining that a dynamic resolution is happening? This seems somewhat hidden this way